CraftingRevisions/BlueprintManager.cs

```diff
@@ -35,7 +37,12 @@ public static void AddBlueprintFromJson(string text, bool validateEarly)
 				throw new ArgumentException("Blueprint text contains no information", nameof(text));
 			}
 
-			AddBlueprint(ModBlueprintData.ParseFromJson(text), validateEarly);
+			// add the blueprint to the HasSet
+			if (!jsonUserBlueprints.Contains(text))
+			{
+				jsonUserBlueprints.Add(text);
+			}
```

Comment:
You don't need to check if it's contained. Hash sets do that automatically in the add method.